### ISSUE TITLE
Tables don't have accessible aria roles

### ISSUE THREAD

**Issue by maxatdetroit** [CityOfDetroit/COD-Design-System#110]
**Describe the bug**

Tables in our UXDS don't have ARIA roles set appropriately.

**To Reproduce**

1. Go to [table docs in UXDS](https://cityofdetroit.github.io/COD-Design-System/?path=%2Fdocs%2Fcomponents-organisms-table--docs) on Firefox
2. Open [the Accessibility Inspector](https://firefox-source-docs.mozilla.org/devtools-user/accessibility_inspector/)
3. Use the click-selector to select the first table in the inspector
4. See that all of the table elements (table, header, body, cell headers, etc.) have a role set to generic.

**Expected behavior**

Each table element should have the corresponding ARIA role set for that element [as defined by W3](https://www.w3.org/TR/html-aam-1.0/#el-table).

**Screenshots**

Current tables:
![Screenshot 2023-10-23 at 4 20 50 PM](https://github.com/CityOfDetroit/COD-Design-System/assets/143553259/88cd6405-4af2-4d9e-9a19-ac7f11ec0190)
